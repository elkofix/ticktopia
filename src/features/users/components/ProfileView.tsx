import { Button } from "@/shared/components/Button";
import { ProfileField } from "./ProfileField";

export const ProfileView = ({ user, onEdit, onOpenModal }: {
  user: { name: string; lastname: string; email: string };
  onEdit: () => void;
  onOpenModal: () => void;
}) => (
  <div className="space-y-6" data-testid="profile-view">
    <ProfileField label="Nombre" value={user.name} />
    <ProfileField label="Apellido" value={user.lastname} />
    <ProfileField label="Email" value={user.email} />
    
    <div className="flex space-x-4 pt-4" data-testid="profile-actions">
      <Button 
        variant="primary" 
        onClick={onEdit}
        data-testid="edit-profile-button"
      >
        Editar Perfil
      </Button>
      <Button 
        variant="danger" 
        onClick={onOpenModal}
        data-testid="close-account-button"
      >
        Cerrar Cuenta
      </Button>
    </div>
  </div>
)