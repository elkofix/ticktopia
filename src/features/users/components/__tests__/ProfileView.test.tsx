import { render, screen, fireEvent } from '@testing-library/react';
import { ProfileView } from '../ProfileView';
// Mock de los componentes hijos para un testing mÃ¡s enfocado
jest.mock('../ProfileField', () => ({
  ProfileField: ({ label, value }: { label: string; value: string }) => (
    <div data-testid={`profile-field-${label.toLowerCase()}`}>
      {label}: {value}
    </div>
  )
}));

jest.mock('../../../../shared/components/Button', () => ({
  Button: ({ children, onClick, variant }: any) => (
    <button 
      data-testid={`button-${variant}`}
      onClick={onClick}
    >
      {children}
    </button>
  )
}));

describe('ProfileView', () => {
  const mockUser = {
    name: 'John',
    lastname: 'Doe',
    email: 'john.doe@example.com'
  };

  const mockOnEdit = jest.fn();
  const mockOnOpenModal = jest.fn();

  const defaultProps = {
    user: mockUser,
    onEdit: mockOnEdit,
    onOpenModal: mockOnOpenModal
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders all user information correctly', () => {
    render(<ProfileView {...defaultProps} />);
    
    expect(screen.getByTestId('profile-field-nombre')).toHaveTextContent('Nombre: John');
    expect(screen.getByTestId('profile-field-apellido')).toHaveTextContent('Apellido: Doe');
    expect(screen.getByTestId('profile-field-email')).toHaveTextContent('Email: john.doe@example.com');
  });

  it('calls onEdit when edit button is clicked', () => {
    render(<ProfileView {...defaultProps} />);
    
    const editButton = screen.getByTestId('button-primary');
    fireEvent.click(editButton);
    
    expect(mockOnEdit).toHaveBeenCalledTimes(1);
  });

  it('calls onOpenModal when close account button is clicked', () => {
    render(<ProfileView {...defaultProps} />);
    
    const closeAccountButton = screen.getByTestId('button-danger');
    fireEvent.click(closeAccountButton);
    
    expect(mockOnOpenModal).toHaveBeenCalledTimes(1);
  });

  it('renders both action buttons', () => {
    render(<ProfileView {...defaultProps} />);
    
    expect(screen.getByText('Editar Perfil')).toBeInTheDocument();
    expect(screen.getByText('Cerrar Cuenta')).toBeInTheDocument();
  });

  it('applies correct spacing classes', () => {
    const { container } = render(<ProfileView {...defaultProps} />);
    
    // Verifica las clases de espaciado
    expect(container.firstChild).toHaveClass('space-y-6');
    const buttonsContainer = screen.getByText('Editar Perfil').parentElement;
    expect(buttonsContainer).toHaveClass('flex', 'space-x-4', 'pt-4');
  });

  it('matches snapshot with default props', () => {
    const { asFragment } = render(<ProfileView {...defaultProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});