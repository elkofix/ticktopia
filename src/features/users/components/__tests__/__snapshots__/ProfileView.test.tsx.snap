// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileView matches snapshot with default props 1`] = `
<DocumentFragment>
  <div
    class="space-y-6"
    data-testid="profile-view"
  >
    <div
      data-testid="profile-field-nombre"
    >
      Nombre: John
    </div>
    <div
      data-testid="profile-field-apellido"
    >
      Apellido: Doe
    </div>
    <div
      data-testid="profile-field-email"
    >
      Email: john.doe@example.com
    </div>
    <div
      class="flex space-x-4 pt-4"
      data-testid="profile-actions"
    >
      <button
        data-testid="button-primary"
      >
        Editar Perfil
      </button>
      <button
        data-testid="button-danger"
      >
        Cerrar Cuenta
      </button>
    </div>
  </div>
</DocumentFragment>
`;
